### PR TITLE
[BUGFIX beta] Throw error if run.bind receives no method

### LINKED ISSUE #16652: Ember.run.bind should throw an error if the callback is undefined

### PR DESCRIPTION
**Pull request by Serabe**
This commit uses the same logic as backburner.join to know
which method is being bound.

Fixes #16652

### CODE REVIEW

File: packages/@ember/runloop/index.js [hunk 1]
```diff
@@ -217,7 +217,39 @@ export function join() {
   @since 1.4.0
   @public
 */
-export const bind = (...curried) => (...args) => join(...curried.concat(args));
+export const bind = (...curried) => {
+  if (!bindMethodPresent(...curried)) {
+    throw new Error('could not find a suitable method to bind');
```
Comment (rwjblue) [inline packages/@ember/runloop/index.js]: Can we make this an assertion and inline the `bindMethodPresent` implementation in an IIFE as the predicate?

```js
assert('Could not find a suitable method to bind.', (function() {
  // guts of conditional below...
})());
```

**Comment by Serabe** [comment]
No problem. I used an error because it was what was requested, but can
easily be done.

**Comment by Serabe** [comment]
Done!

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
Thank you!!

**Pull request merged**